modules/imgcodecs/src/grfmt_tiff.cpp

```diff
@@ -559,6 +564,35 @@ bool TiffDecoder::readHdrData(Mat& img)
     return true;
 }
 
+bool TiffDecoder::readData_32FC1(Mat& img)
+{
+    if(!m_tif)
+    {
+        return false;
+    }
+    TIFF *tif = static_cast<TIFF*>(m_tif);
+
+    uint32 img_width, img_height;
+    TIFFGetField(tif,TIFFTAG_IMAGEWIDTH, &img_width);
+    TIFFGetField(tif,TIFFTAG_IMAGELENGTH, &img_height);
+    tsize_t scanlength = TIFFScanlineSize(tif);
+    tdata_t buf = _TIFFmalloc(scanlength);
+    float* data;
+    for (uint32 row = 0; row < img_height; row++)
+    {
+        TIFFReadScanline(tif, buf, row);
```

Comment:
[TIFFReadScanline returns -1 if it detects an error; otherwise 1 is returned](http://www.libtiff.org/man/TIFFReadScanline.3t.html)

BTW, Don't forget to free resources during error handling.